recipes/_windows.rb

```diff
@@ -16,98 +16,70 @@
-  msys_p('http://downloads.sourceforge.net/mingw/libregex-1.20090805-2-msys-1.0.13-dll-1.tar.lzma',
-    '85dd8c1e27a90675c5f867be57ba7ae2bb55dde8cd2d19f284c896be134bd3d1'),
-  msys_p('http://downloads.sourceforge.net/mingw/crypt-1.1_1-3-msys-1.0.13-bin.tar.lzma',
-    '58369b42c38144d3aa5a337ebf1e182a66e88db30ccc42796f2074f251ee1fed'),
-  msys_p('http://downloads.sourceforge.net/mingw/m4-1.4.14-1-msys-1.0.13-bin.tar.lzma',
-    '41058bc9a691ad01fdd979f1a4ac4ee071bd5ce93f660db5c0b3cfad4487e33e'),
-  msys_p('http://downloads.sourceforge.net/mingw/bison-2.4.2-1-msys-1.0.13-bin.tar.lzma',
-    '349f3e312bf71f8a2ac68a7bd2f86b03dacc565b0fd27eef5d604e8be402390e'),
-  msys_p('http://downloads.sourceforge.net/mingw/flex-2.5.35-2-msys-1.0.13-bin.tar.lzma',
-    '9715511a2eafb7e2402029059d4b9db96bd40d8b72908db2571c009745c47a63'),
-  msys_p('http://downloads.sourceforge.net/mingw/findutils-4.4.2-2-msys-1.0.13-bin.tar.lzma',
-    '779e819b7942dc070c45f4cba633e6a9ae4bfe8b506a3541f4ce86ad0595726d'),
-  msys_p('http://downloads.sourceforge.net/mingw/sed-4.2.1-2-msys-1.0.13-bin.tar.lzma',
-    'f73059204cecb691e7840108b7c0cbbfcebf50c0e5c6e3a2326e0eedce5d1b94'),
-  msys_p('http://downloads.sourceforge.net/mingw/gawk-3.1.7-2-msys-1.0.13-bin.tar.lzma',
-    'eb15478ea76e75b666ad7fc7049de21b9f487e0e1ea0e96d40953a477e91c3dd'),
-  msys_p('http://downloads.sourceforge.net/mingw/grep-2.5.4-2-msys-1.0.13-bin.tar.lzma',
-    '4842a1754df98db994622e8ffab3bea7fbce77e05778cd5d3831e76ac90440ba'),
-  msys_p('http://downloads.sourceforge.net/mingw/less-436-2-msys-1.0.13-bin.tar.lzma',
-    '1bbd114846026f9ca4fcc4e18ba20f060384f623f1ef22b326df8c55419c0b84'),
-  msys_p('http://downloads.sourceforge.net/mingw/diffutils-2.8.7.20071206cvs-3-msys-1.0.13-bin.tar.lzma',
-    '522889b044492dd2337c4752ba6262995a11f352ca5fb8a8660349413ea9b864'),
-  msys_p('http://downloads.sourceforge.net/mingw/texinfo-4.13a-2-msys-1.0.13-bin.tar.lzma',
-    '241eb8e376bf69588d0e02aede35771503c5dcb15c440f97e15e30da79fea864'),
-  msys_p('http://downloads.sourceforge.net/mingw/libmagic-5.04-1-msys-1.0.13-dll-1.tar.lzma',
-    '65117008598675823b3fb25296d0d6c332ce56b72950e0f90f9063ac098afac3'),
-  msys_p('http://downloads.sourceforge.net/mingw/file-5.04-1-msys-1.0.13-bin.tar.lzma',
-    'e9ceffa49629524c84d07da77c1a5f37837f68a09e56cad30bea1df0a21e5fc2'),
-  msys_p('http://downloads.sourceforge.net/mingw/mintty-1.0.3-1-msys-1.0.17-bin.tar.lzma',
-    '0b3e7b57c81646eccaff3ca0310abe8367ace69992640be87199ecf5d9443085'),
-  msys_p('http://downloads.sourceforge.net/mingw/patch-2.6.1-1-msys-1.0.13-bin.tar.lzma',
-    'c8b7771304fb5e9fc33d8fca9045402f2e1bca055bf0b28127f3c3e85a254f67')
-].each do |package|
+tool32_path = node['build-essential']['mingw32']['path'] if node['build-essential']['mingw32']
+tool64_path = node['build-essential']['mingw64']['path'] if node['build-essential']['mingw64']
+tool32_path = nil if tool32_path == ''
+tool64_path = nil if tool64_path == ''
```

Comment:
We can really just rely on `node['build-essential']['mingw*']['path']` being set to a real value here, that's the whole point of default values for attributes.